### PR TITLE
Move and namespaces legacy fields

### PR DESCRIPTION
**Pull request by fastslack**
### Summary of Changes

This change move legacy form fields to Joomla\CMS\Form\Field directory for namespaces

### Testing Instructions

Use Category , ComponentLayout or ModuleLayout form field and should work as usual

### Expected result

Work as usual

### CODE REVIEW

File: libraries/src/Joomla/CMS/Form/Field/CategoryField.php [hunk 1]
```diff
@@ -82,7 +86,7 @@ protected function getOptions()
 					 * unless the item is already in that category.
 					 * Unset the option if the user isn't authorised for it. In this field assets are always categories.
 					 */
-					if ($user->authorise('core.create', $extension . '.category.' . $option->value) === false)
+					if ($user->authorise('core.create', $extension . '.category.' . $option->value) != true)
```
Comment (C-Lodder) [inline libraries/src/Joomla/CMS/Form/Field/CategoryField.php]: `!==`

Comment (fastslack) [reply libraries/src/Joomla/CMS/Form/Field/CategoryField.php]: done

**Review by mbabker — CHANGES_REQUESTED** [review]
Needs class mapping.

**Comment by wilsonge** [comment]
Do we want the category form field to be moved into the category component rather than being a library form field?

File: libraries/src/Joomla/CMS/Form/Field/ComponentlayoutField.php [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+namespace Joomla\CMS\Form\Field;
+
 defined('JPATH_PLATFORM') or die;
 
-jimport('joomla.filesystem.folder');
+use Joomla\CMS\Application\ApplicationHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Form\Form;
+use Joomla\CMS\Form\FormField;
+use Joomla\CMS\Filesystem\Folder;
```
Comment (wilsonge) [inline libraries/src/Joomla/CMS/Form/Field/ComponentlayoutField.php:17]: These haven't been namespaced yet

File: libraries/src/Joomla/CMS/Form/Field/ModulelayoutField.php [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+namespace Joomla\CMS\Form\Field;
+
 defined('JPATH_PLATFORM') or die;
 
-jimport('joomla.filesystem.folder');
+use Joomla\CMS\Application\ApplicationHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Form\Form;
+use Joomla\CMS\Form\FormField;
+use Joomla\CMS\Filesystem\Folder;
```
Comment (wilsonge) [inline libraries/src/Joomla/CMS/Form/Field/ModulelayoutField.php:17]: Same here

**Review by wilsonge — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Status is set on "Needs Review".

**Comment by fastslack** [comment]
Should I do same thing but for 3.9 branch and close this PR?

**Comment by Quy** [comment]
@wilsonge Close PR?

File: libraries/src/Joomla/CMS/Form/Field/CategoryField.php [hunk 4]
```diff
@@ -1,23 +1,27 @@
 <?php
 /**
- * @package     Joomla.Legacy
- * @subpackage  Form
+ * Joomla! Content Management System
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
```
Comment (PhilETaylor) [inline libraries/src/Joomla/CMS/Form/Field/CategoryField.php:5]:
```suggestion
 * @copyright  (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

File: libraries/src/Joomla/CMS/Form/Field/ComponentlayoutField.php [hunk 5]
```diff
@@ -1,31 +1,37 @@
 <?php
 /**
- * @package     Joomla.Legacy
- * @subpackage  Form
+ * Joomla! Content Management System
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
```
Comment (PhilETaylor) [inline libraries/src/Joomla/CMS/Form/Field/ComponentlayoutField.php:5]:
```suggestion
 * @copyright  (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

File: libraries/src/Joomla/CMS/Form/Field/ModulelayoutField.php [hunk 6]
```diff
@@ -1,30 +1,36 @@
 <?php
 /**
- * @package     Joomla.Legacy
- * @subpackage  Form
+ * Joomla! Content Management System
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
```
Comment (PhilETaylor) [inline libraries/src/Joomla/CMS/Form/Field/ModulelayoutField.php:5]:
```suggestion
 * @copyright  (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by laoneo** [comment]
Sorry that it has taken so long to respond to this pull request. We did this change in Joomla 4 already and wont move the files in Joomla 3. So I'm closing this pr. Nevertheless we would like to thank you for your contribution.

**Pull request closed without merging**